pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -1224,7 +1226,23 @@ public ShutdownService getShutdownService() {
      * @return Hostname or IP address the service advertises to the outside world.
      */
     public static String advertisedAddress(ServiceConfiguration config) {
-        return ServiceConfigurationUtils.getDefaultOrConfiguredAddress(config.getAdvertisedAddress());
+        String advertisedAddress = null;
+        if (!isBlank(config.getAdvertisedAddress())) {
+            advertisedAddress = config.getAdvertisedAddress();
+        } else {
+            String iface = config.getListeningInterface();
+            if (!isBlank(iface)) {
+                try {
+                    advertisedAddress = DNS.getDefaultIP(iface);
+                } catch (UnknownHostException e) {
+                    LOG.error("GetAdvertisedAddressIP Fail: {}", iface);
```

Comment:
Do we need to throw an exception or use a default value if the error occurs?

Reply:
Advertised address uses localhost as the default?

code:
        String advertisedAddress = “localhost”;
        if (!isBlank(config.getAdvertisedAddress())) {
            advertisedAddress = config.getAdvertisedAddress().trim();
        } else {
            String iface = config.getListeningInterface();
            if (!isBlank(iface)) {
                try {
                    advertisedAddress = DNS.getDefaultIP(iface);
                } catch (UnknownHostException e) {
                    LOG.error("GetAdvertisedAddressIP Fail: {}", iface);
                }
                LOG.info("iface advertisedAddress: {}", advertisedAddress);
            } else {
                advertisedAddress =
                        ServiceConfigurationUtils.unsafeLocalhostResolve();
                LOG.info("ServiceConfigurationUtils hostname: {}", advertisedAddress);
            }

        }
        return advertisedAddress;

Reply:
> Do we need to throw an exception or use a default value if the error occurs?

Advertised address uses localhost as the default?

code:
String advertisedAddress = “localhost”;
if (!isBlank(config.getAdvertisedAddress())) {
advertisedAddress = config.getAdvertisedAddress().trim();
} else {
String iface = config.getListeningInterface();
if (!isBlank(iface)) {
try {
advertisedAddress = DNS.getDefaultIP(iface);
} catch (UnknownHostException e) {
LOG.error("GetAdvertisedAddressIP Fail: {}", iface);
}
LOG.info("iface advertisedAddress: {}", advertisedAddress);
} else {
advertisedAddress =
ServiceConfigurationUtils.unsafeLocalhostResolve();
LOG.info("ServiceConfigurationUtils hostname: {}", advertisedAddress);
}

    }
    return advertisedAddress;